src/executor/source/executor/Linearization.ts

```diff
@@ -1,11 +1,11 @@
-import { Dataflow, Edge, LinearizedDataflow, Node } from '../Dataflow'
+import { Dataflow, Edge, Node } from '../Dataflow'
 
 /**
  * @param dataflow to order in such a way that a iterative execution of the nodes
  * is possible where all inputs are calculated at the time of the execution
  * @returns the linearized dataflow
  */
-export function linearize( dataflow: Dataflow ): LinearizedDataflow {
+export async function linearize( dataflow: Dataflow ): Promise<any> {
```

Comment:
- Why is this function async? There is no await or similar in its body.
- Is the any-type in the Promise really necessary or can this be typed even stronger? The function should return a LinearizedDataflow-object only right?